### PR TITLE
Create typings for the parse-listing library

### PR DESCRIPTION
**Pull request by maximelebastard**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## parse-listing ([<kbd>unpkg</kbd>](https://unpkg.com/browse/parse-listing@latest/))
was missing the following properties:
1. nodeTypes
1. permissions
1. access
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 97f460625256bcd5444ef86ec4a3f5244c6be663
</p>

File: types/parse-listing/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export interface Permissions {
+    read: boolean;
+    write: boolean;
+    exec: boolean;
+}
+
+export type ParsingCallback = (err: Error, result: File[]) => void;
+
+export enum NODE_TYPES {
```
Comment (gabritto) [inline types/parse-listing/index.d.ts:26]:
```suggestion
export enum nodeTypes {
```

File: types/parse-listing/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    FILE_TYPE = 0,
+    DIRECTORY_TYPE = 1,
+    SYMBOLIC_LINK_TYPE = 2,
+    UNKNOWN_TYPE = 3,
+}
+export enum PERMISSIONS {
```
Comment (gabritto) [inline types/parse-listing/index.d.ts:32]:
```suggestion
export enum permissions {
```

File: types/parse-listing/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    READ_PERMISSION = 0,
+    WRITE_PERMISSION = 1,
+    EXECUTE_PERMISSION = 2,
+}
+export enum ACCESS {
```
Comment (gabritto) [inline types/parse-listing/index.d.ts:37]:
```suggestion
export enum access {
```

**Review by gabritto — CHANGES_REQUESTED** [review]
seems good to me, except for the names of the enums (based on https://unpkg.com/browse/parse-listing@1.1.3/src/parser.js, instead of e.g. `NODE_TYPES` it should be `nodeTypes`).

**Pull request closed without merging**